src/test/java/io/supertokens/test/userIdMapping/UserIdMappingStorageTest.java

```diff
@@ -341,4 +341,164 @@ public void testRetrievingUserIdMapping() throws Exception {
+            assertNull(storage.getUserIdMapping(superTokensUserId, true));
+
+        }
+
+        {
+            // create a new userId mapping
+            String newExternalUserId = "externalUserIdNew";
+            storage.createUserIdMapping(superTokensUserId, newExternalUserId, null);
+            // retrieve mapping and check that it exists
+            UserIdMapping userIdMapping = storage.getUserIdMapping(newExternalUserId, false);
+            assertNotNull(userIdMapping);
+            assertEquals(superTokensUserId, userIdMapping.superTokensUserId);
+            assertEquals(newExternalUserId, userIdMapping.externalUserId);
+            assertNull(userIdMapping.externalUserIdInfo);
+
+            // delete mapping with externalUserId
+            assertTrue(storage.deleteUserIdMapping(newExternalUserId, false));
+
+            // check that the mapping does not exist
+            assertNull(storage.getUserIdMapping(newExternalUserId, false));
+
+        }
+
+        process.kill();
+        assertNotNull(process.checkOrWaitForEvent(ProcessState.PROCESS_STATE.STOPPED));
+    }
+
+    @Test
+    public void testUpdatingExternalUserIdWithUnknownUserId() throws Exception {
+        String[] args = { "../" };
+        TestingProcessManager.TestingProcess process = TestingProcessManager.start(args);
+        assertNotNull(process.checkOrWaitForEvent(ProcessState.PROCESS_STATE.STARTED));
+
+        if (StorageLayer.getStorage(process.getProcess()).getType() != STORAGE_TYPE.SQL) {
+            return;
+        }
+
+        UserIdMappingStorage storage = StorageLayer.getUserIdMappingStorage(process.main);
+
+        // update with unknown supertokensUserId
+        assertFalse(storage.updateOrDeleteExternalUserIdInfo("unknownId", true, null));
+
+        // update with unknown externalUserId
+        assertFalse(storage.updateOrDeleteExternalUserIdInfo("unknownId", false, null));
+
+        process.kill();
+        assertNotNull(process.checkOrWaitForEvent(ProcessState.PROCESS_STATE.STOPPED));
+    }
+
+    @Test
+    public void testUpdatingExternalUserId() throws Exception {
+        String[] args = { "../" };
+        TestingProcessManager.TestingProcess process = TestingProcessManager.start(args);
+        assertNotNull(process.checkOrWaitForEvent(ProcessState.PROCESS_STATE.STARTED));
+
+        if (StorageLayer.getStorage(process.getProcess()).getType() != STORAGE_TYPE.SQL) {
+            return;
+        }
+
+        UserIdMappingStorage storage = StorageLayer.getUserIdMappingStorage(process.main);
+
+        // create User
+        UserInfo userInfo = EmailPassword.signUp(process.main, "test@example.com", "testPass123");
+
+        String superTokensUserId = userInfo.id;
+        String externalUserId = "externalId";
+        String externalUserIdInfo = "externalUserIdInfo";
+
+        // create a userId mapping
+        storage.createUserIdMapping(superTokensUserId, externalUserId, externalUserIdInfo);
```

Comment:
Add a different test where you are creating it with `null` as the externalUserIdInfo and see that when you retrieve it, it gives back null